src/bridge/bridge.c

```diff
@@ -468,23 +469,31 @@ run_bridge (const gchar *interactive,
   GPid agent_pid = 0;
   guint sig_term;
   guint sig_int;
-  int outfd;
   uid_t uid;
   struct CallUpdateRouterData call_update_router_data;
 
-  /*
-   * This process talks on stdin/stdout. However lots of stuff wants to write
-   * to stdout, such as g_debug, and uses fd 1 to do that. Reroute fd 1 so that
-   * it goes to stderr, and use another fd for stdout.
-   */
+  if (outfd < 0)
+    {
+      g_printerr ("file descriptors must not be negative\n");
+      return 1;
+    }
 
-  outfd = dup (1);
-  if (outfd < 0 || dup2 (2, 1) < 1)
+  if (outfd == 1)
     {
-      g_printerr ("bridge couldn't redirect stdout to stderr");
-      if (outfd > -1)
-        close (outfd);
-      outfd = 1;
+      /*
+       * This process talks on stdin/stdout. However lots of stuff wants to write
+       * to stdout, such as g_debug, and uses fd 1 to do that. Reroute fd 1 so that
+       * it goes to stderr, and use another fd for stdout.
```

Comment:
On second read, this comment isn't true (and never was).  We are indeed rerouting fd1 to stderr (which might be more clearly written as "dup stderr to also be used for stdout output") but we are definitely not "us[ing] another fd for stdout" → we're actually using another fd for cockpit's stream output.

Reply:
Fixed the comment, thanks for pointing out!